### ISSUE TITLE
[BUG] PUT _index_template/mytemplate API failed (403 Forbidden) when setting correct permissions

### ISSUE THREAD

**Issue by oridool** [opensearch-project/security#1900]
**What is the bug?**
I have a user with limited privileges (applying to least privileges pattern). 
When calling the API 
`PUT _index_template/my_template  { ... }'

It failed with 403 even though the correct permission is applied. 
The error message is:
```
Caused by: org.elasticsearch.client.ResponseException: method [PUT], host [https://x.x.x.x:9200], URI [_index_template/m_template], status line [HTTP/1.1 403 Forbidden]
{"error":{"root_cause":[{"type":"security_exception","reason":"no permissions for [indices:admin/index_template/put] and User [name=ori_management, backend_roles=[ori_management], requestedTenant=null]"}],"type":"security_exception","reason":"no permissions for [indices:admin/index_template/put] and User [name=ori_management, backend_roles=[ori_management], requestedTenant=null]"},"status":403}
```

**How can one reproduce the bug?**

create a user with the following cluster permissions:
```
                    "cluster_permissions": [
						"cluster_monitor",
						"indices:data/write/bulk"
						"indices:data/read/scroll"
						"indices:admin/template/put"
						"indices:admin/index_template/put"
						"cluster:admin/component_template/put"
                    ]
```

Execute the `PUT _index_template` API.
It failed with 403 - Forbidden.

**What is the expected behavior?**
Since the user has the `indices:admin/index_template/put` permission, the API should succeed.

**What is your host/environment?**
 - OS: Centos 7
 - Version Opensearch 1.3.3
 - Plugins 
```
 sh-4.2$ bin/opensearch-plugin list
opensearch-alerting
opensearch-anomaly-detection
opensearch-asynchronous-search
opensearch-cross-cluster-replication
opensearch-index-management
opensearch-job-scheduler
opensearch-knn
opensearch-observability
opensearch-performance-analyzer
opensearch-reports-scheduler
opensearch-security
opensearch-sql
```

**Comment by oridool** [comment]
In the meanwhile, I have found a workaround, as provided in this thread:
https://forum.opensearch.org/t/unable-to-set-permissions-for-put-index-template/4863/9 

> There is a bug it seems, where opensearch doesn’t read the cluster permission indices:admin/index_template/put from the cluster permission section (I’d recommend to raise a ticket for this on [github 2](https://github.com/opensearch-project/security)), however a quick workaround should be to apply only this permission to all indices (*)

**Comment by DarshitChanpura** [comment]
[Triage] This fix has been implemented to 2.0 and so we would suggest you to upgrade. If it's not a viable option, a backport to 1.3 and wait for its scheduled release as a plan B.

**Comment by peternied** [comment]
[Triage] Closing out this issue as the fix has been merged and released in 2.1
